qadence/constructors/id_init_ansatz.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from typing import Any, Type, Union
+
+import torch
+
+from qadence.blocks import AbstractBlock, chain, kron, tag
+from qadence.operations import CNOT, CRX, CRY, CRZ, CZ, RX, RY
+from qadence.parameters import VariationalParameter
+
+DigitalEntanglers = Union[CNOT, CZ, CRZ, CRY, CRX]
+
+
+def _entangler(
+    control: int,
+    target: int,
+    param_str: str,
+    op: Type[DigitalEntanglers] = CNOT,
+) -> AbstractBlock:
+    if op in [CNOT, CZ]:
+        return op(control, target)  # type: ignore
+    elif op in [CRZ, CRY, CRX]:
+        param = VariationalParameter(param_str, value=0.0)
+        return op(control, target, param)  # type: ignore
+    else:
+        raise ValueError("Provided entangler not accepted for digital ansatz.")
+
+
+def identity_initialized_ansatz(
+    n_qubits: int,
+    depth: int = 1,
+    rotations: list[type[AbstractBlock]] = [RX, RY],
+    entangler: Any = CNOT,
+    periodic: bool = False,
+) -> AbstractBlock:
+    """
+    Identity block for barren plateau mitigation.
+
+    The initial configuration of this block is equal to an identity unitary
+    but can be trained in the same fashion as other ansatzes, reaching same level
+    of expressivity.
+
+    Args:
+        n_qubits: number of qubits in the block
+        depth: number of layers of the HEA
+        rotations (list of AbstractBlocks):
+            single-qubit rotations with trainable parameters
+        entangler (AbstractBlock):
+            2-qubit entangling operation. Supports CNOT, CZ, CRX, CRY, CRZ, CPHASE.
+            Controlled rotations will have variational parameters on the rotation angles.
+        periodic (bool): if the qubits should be linked periodically.
+    """
+    initialized_layers = []
+    for layer in range(depth):
+        alpha = 2 * torch.pi * torch.rand(n_qubits * len(rotations))
+        gamma = torch.zeros(n_qubits)
+        beta = -alpha
+        left_rotations = [
```

Comment:
My only nitpick is if you could somehow write the code in this function a bit more structured and in shorter lines, with less chaining of operations. It's very annoying I know 😅 but because we have a limit on line length, the pre-commit hooks will format it like it is now, which makes it harder to read.

Reply:
So e.g. instead of `kron(stuff for i in range(n_qubits))` putting the `for` in a separate line above. You can use list appends for example and then at the end do `kron(*list)`.

Reply:
Sounds good, @jpmoutinho. I reorganized the way left and right rotations are created, let me know if this version accomplishes what you pointed out.

Reply:
Thanks! It looks better :)